src/features/swap/hooks/useStaking.ts

```diff
@@ -0,0 +1,63 @@
+import BigNumber from 'bignumber.js';
+import { useCallback, useMemo, useState } from 'react';
+import { GAS_PRICE } from 'src/config/consts';
+import { useAccountContext } from 'src/contexts/account/AccountContext';
+import { useProtocolContext } from 'src/contexts/protocol/ProtocolContext';
+import { TxCallbacks, useBlockchain } from 'src/hooks/useBlockchain';
+import { Celo, CeloUSD, StCelo } from 'src/utils/tokens';
+import { showStakingToast } from '../utils/toast';
+
+export function useStaking() {
+  const { address, loadBalances, celoBalance } = useAccountContext();
+  const { managerContract, stCeloContract, sendTransaction } = useBlockchain();
+  const { stakingRate, celoToUSDRate } = useProtocolContext();
+  const [celoAmount, setCeloAmount] = useState<Celo | null>(null);
+
+  const createTxOptions = useCallback(
+    () => ({
+      from: address!,
```

Comment:
Application flow makes it impossible to reach this place without address being set (if user doesn't have an address it means he's not connected to the wallet, hence the application will redirect him to `/connect` page). 

However, I prefer not to have code with `!` - this is a form of cheating in TypeScript ;) 
I decided to change this method in such a way that it will throw an error in case `address` is undefined, even though application should never be in such state.